src/main/java/edu/cornell/kfs/fp/batch/service/impl/AwsAccountingXmlDocumentAccountingLineServiceImpl.java

```diff
@@ -41,222 +43,268 @@
-        }
-
-        ProjectCode projectCode = projectCodeService.getByPrimaryId(projectCodeValue);
-        if (ObjectUtils.isNull(projectCode)) {
-            LOG.error(String.format(configurationService.getPropertyValueAsString(CuFPKeyConstants.ERROR_PROJECT_CODE_NOT_FOUND), projectCodeValue));
-            return false;
-        }
-        if (!projectCode.isActive()) {
-            LOG.error(String.format(configurationService.getPropertyValueAsString(CuFPKeyConstants.ERROR_PROJECT_CODE_INACTIVE), projectCodeValue));
-            return false;
-        }
-        return true;
+    private boolean validateProjectCode(String projectCodeValue, boolean logErrorMessage) {
+        boolean valid = true;
+        if (StringUtils.isNotBlank(projectCodeValue)) {
+            ProjectCode projectCode = projectCodeService.getByPrimaryId(projectCodeValue);
+            if (ObjectUtils.isNull(projectCode)) {
+                logAccountValidationError(String.format(configurationService.getPropertyValueAsString(CuFPKeyConstants.ERROR_PROJECT_CODE_NOT_FOUND), projectCodeValue), logErrorMessage);
+                valid = false;
+            } else if (!projectCode.isActive()) {
+                LOG.error(String.format(configurationService.getPropertyValueAsString(CuFPKeyConstants.ERROR_PROJECT_CODE_INACTIVE), projectCodeValue));
+                valid = false;
+            }
+        }
+        return valid;
     }
 
-    private boolean validateOrgRefId(String orgRefId) {
-        if (StringUtils.isBlank(orgRefId)) {
-            return true;
-        }
-        if (orgRefId.length() > 8) {
-            LOG.error(String.format(configurationService.getPropertyValueAsString(CuFPKeyConstants.ERROR_ORG_REF_ID_TOO_LONG), orgRefId));
+    private boolean validateOrgRefId(String orgRefId, boolean logErrorMessage) {
+        if (StringUtils.isNotBlank(orgRefId) && orgRefId.length() > 8) {
+            logAccountValidationError(String.format(configurationService.getPropertyValueAsString(CuFPKeyConstants.ERROR_ORG_REF_ID_TOO_LONG), orgRefId), logErrorMessage);
             return false;
         }
         return true;
     }
-
-    private AccountingXmlDocumentAccountingLine getDefaultAccountingLine(String defaultAccountString) {
-        AccountingXmlDocumentAccountingLine xmlAccountingLine = getEmptyAccountingXmlDocumentAccountingLine();
-
-        if (StringUtils.countMatches(defaultAccountString, "*") == 6) {
-            xmlAccountingLine = parseCostCenterGroupLevel(defaultAccountString);
-        }
-        else if (StringUtils.countMatches(defaultAccountString, "-") == 1) {
-            xmlAccountingLine.setAccountNumber(StringUtils.split(defaultAccountString, "-")[0]);
-            xmlAccountingLine.setSubAccountNumber(StringUtils.split(defaultAccountString, "-")[1]);
-        }
-        else {
-            xmlAccountingLine.setAccountNumber(defaultAccountString);
+    
+    private void logAccountValidationError(String message, boolean logErrorMessage) {
```

Comment:
Could you help me understand why sometimes we want to log the error message and other times we don't?

Reply:
We run the validation multiple times.  Once to identify default accounts or cost centers that have errors.  The second time is when the actual transaction lines are being built.  We only want to log once so we don't have duplicate log statements.

The validation routines run very quickly and doesn't cause a performance issue running them more than once.